sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -876,6 +913,18 @@ public Read withHighPriority() {
       return withSpannerConfig(config.withRpcPriority(RpcPriority.HIGH));
     }
 
+    private SpannerSourceDef createSourceDef() {
+      SpannerSourceDef sourceDef;
+      if (getReadOperation().getQuery() != null) {
+        sourceDef = SpannerQuerySourceDef.create(getSpannerConfig(), getReadOperation().getQuery());
+      } else {
+        sourceDef =
+            SpannerTableSourceDef.create(
+                getSpannerConfig(), getReadOperation().getTable(), getReadOperation().getColumns());
+      }
+      return sourceDef;
```

Comment:
Nit: Can be cleaned up a bit with an early return

```suggestion
      if (getReadOperation().getQuery() != null) {
        return SpannerQuerySourceDef.create(getSpannerConfig(), getReadOperation().getQuery());
      }
      
      return SpannerTableSourceDef.create(
          getSpannerConfig(), getReadOperation().getTable(), getReadOperation().getColumns());;
```

Reply:
Done.